src/libmongoc/tests/test-mongoc-background-monitoring.c

```diff
@@ -600,16 +683,441 @@ test_repeated_requestscan (void)
    request_t *request;
    int i;
 
+   /* Multiple repeated requests before an ismaster completes should not cause a
+    * subsequent scan. */
    tf = tf_new (TF_FAST_MIN_HEARTBEAT);
-
-   for (i = 1; i < 100; i++) {
-      request = mock_server_receives_ismaster (tf->server);
-      OBSERVE (tf, request);
-      mock_server_replies_ok_and_destroys (request);
-      OBSERVE_SOON (tf, tf->observations->n_heartbeat_started == i);
-      OBSERVE_SOON (tf, tf->observations->n_heartbeat_succeeded == i);
+   for (i = 0; i < 10; i++) {
+      _request_scan (tf);
+   }
+   request = mock_server_receives_ismaster (tf->server);
+   OBSERVE (tf, request);
+   for (i = 0; i < 10; i++) {
       _request_scan (tf);
    }
+   WAIT;
+   OBSERVE (tf, tf->observations->n_heartbeat_started == 1);
+   mock_server_replies_ok_and_destroys (request);
+   OBSERVE_SOON (tf, tf->observations->n_heartbeat_succeeded == 1);
+
+   tf_destroy (tf);
+}
+
+static void
+test_sleep_after_scan (void)
+{
+   test_fixture_t *tf;
+   request_t *request;
+
+   /* After handling a scan request */
+   tf = tf_new (TF_FAST_MIN_HEARTBEAT);
+   _request_scan (tf);
+   request = mock_server_receives_ismaster (tf->server);
+   OBSERVE (tf, tf->observations->n_heartbeat_started == 1);
+   mock_server_replies_ok_and_destroys (request);
+   OBSERVE_SOON (tf, tf->observations->n_heartbeat_succeeded == 1);
+   /* Wait for two min heartbeat periods. */
+   WAIT;
```

Comment:
Per my earlier suggestion, a more descriptive macro would also make the comment unnecessary.